jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/IOResources.java

```diff
@@ -90,6 +90,20 @@ public static RetainableByteBuffer toRetainableByteBuffer(Resource resource, Byt
             }
         }
 
+        // Optimize for Content.Source.Factory.
+        if (resource instanceof Content.Source.Factory factory)
+        {
+            try
+            {
+                ByteBuffer buffer = Content.Source.asByteBuffer(factory.newContentSource(bufferPool, 4096, direct, 0L, -1L));
```

Comment:
`-1` should also be a constant, e.g., `ENTIRE_SIZE` or something.

Reply:
I think it's an anti-pattern already to reuse the same methods for the case where an entire stream is consumed vs. a range. (I'm not sure if we even have the use for offset != 0 but limit = -1). 

If we would consistently keep at least the public methods of "everything" vs "this range" separate (as we already partially do in IOResources), we wouldn't even need to think about it.

Feel free to ignore this suggestion.